src/main/java/org/jabref/logic/journals/PredatoryJournalLoader.java

```diff
@@ -0,0 +1,145 @@
+        try (InputStream resourceAsStream = PredatoryJournalRepository.class.getResourceAsStream("/journals/predatoryJournal-list.mv")) {
+            if (resourceAsStream == null) {
+                LOGGER.warn("There is no predatoryJournalList.mv. We use a default predatory journal list");
+            } else {
+                // Use user's app data directory for more permanent storage
+                Path appDataDir = Path.of(AppDirsFactory.getInstance()
+                                                        .getUserDataDir(
+                                                                OS.APP_DIR_APP_NAME,
+                                                                "predatoryJournals",
+                                                                OS.APP_DIR_APP_AUTHOR));
+                Files.createDirectories(appDataDir); // Ensure the directory exists
+                Path predatoryJournalListPath = appDataDir.resolve("predatoryJournal-list.mv");
+                Files.copy(resourceAsStream, predatoryJournalListPath, StandardCopyOption.REPLACE_EXISTING);
+                loader.repository = new PredatoryJournalRepository(predatoryJournalListPath);
+            }
+        } catch (IOException e) {
+            LOGGER.error("Error while copying predatory journal list", e);
+            return null;
+        }
+        return loader.repository;
+    }
+
+    public void update() {
+        // populates linkElements (and predatoryJournals if CSV)
+        PREDATORY_SOURCES.forEach(this::crawl);
+        // adds cleaned HTML to predatoryJournals
+        LINK_ELEMENTS.forEach(this::clean);
+
+        LOGGER.info("Updated predatory journal list");
+    }
+
+    private void crawl(PJSource source) {
+        try {
+            URLDownload download = new URLDownload(source.url);
+
+            if (!download.canBeReached()) {
+                LOGGER.warn("URL UNREACHABLE");
+            } else if (source.url.getPath().contains(".csv")) {
+                handleCSV(new InputStreamReader(download.asInputStream()));
+            } else {
+                handleHTML(source.elementPattern, download.asString());
+            }
+        } catch (Exception ex) {
+            LOGGER.error("Could not crawl source {}", source.url, ex);
+        }
+    }
+
+    private void handleCSV(Reader reader) throws IOException {
+        CSVParser csvParser = new CSVParser(reader, CSVFormat.EXCEL);
+
+        for (CSVRecord csvRecord : csvParser) {
+            // changes column order from CSV (source: url, name, abbr)
+            repository.addToPredatoryJournals(csvRecord.get(1), csvRecord.get(2), csvRecord.get(0));
+        }
+    }
+
+    private void handleHTML(Pattern pattern, String body) {
+        Matcher matcher = pattern != null ? pattern.matcher(body) : null;
+
+        if (matcher != null) {
+            while (matcher.find()) {
+                LINK_ELEMENTS.add(matcher.group());
+            }
+        }
```

Comment:
Strange code. Please apply fail fast!

```suggestion
        if (pattern == null) {
            return;
        }
        Matcher matcher = pattern.matcher(body);
        while (matcher.find()) {
            LINK_ELEMENTS.add(matcher.group());
        }
```

Moreover, why can't the cleaning the done here?

Please make `handleHTML` consistent to `handleCSV`: The result of `handleHTML`  is to populate the `repository` - and not to populate some internal data structurres.